Sources/BraintreeVisaCheckout/BTVisaCheckoutClient.swift

```diff
@@ -117,16 +112,23 @@ import BraintreeCore
         completion: @escaping (BTVisaCheckoutNonce?, Error?) -> Void
     ) {
         if statusCode == .statusUserCancelled {
+            _ = NSError(
+                domain: BTVisaCheckoutError.errorDomain,
+                code: BTVisaCheckoutError.canceled.errorCode,
+                userInfo: [
+                    NSLocalizedDescriptionKey:
+                        BTVisaCheckoutError.canceled.errorDescription ?? "Visa Checkout flow was canceled by the user."
+                ]
+            )
             notifyFailure(with: BTVisaCheckoutError.canceled, completion: completion)
             return
         }
 
-        guard statusCode == .statusSuccess else {
-            notifyFailure(with: BTVisaCheckoutError.checkoutUnsuccessful, completion: completion)
-            return
-        }
-
-        guard let callID, let encryptedKey, let encryptedPaymentData else {
+        guard
+            let callID = callID,
+            let encryptedKey = encryptedKey,
+            let encryptedPaymentData = encryptedPaymentData
```

Comment:
Suggestion: since we’re using Swift 5.7+, we could simplify the optional binding with the shorthand syntax to keep the code cleaner:

```suggestion
guard let callID,
      let encryptedKey,
      let encryptedPaymentData else { return }

Reply:
I kept it in